src/FRP/Event.purs

```diff
@@ -103,6 +115,17 @@ foreign import sampleOn :: forall a b. Event a -> Event (a -> b) -> Event b
 sampleOn_ :: forall a b. Event a -> Event b -> Event a
 sampleOn_ a b = sampleOn a (b $> id)
 
+-- | Map over an event with an accumulator value. This can be used, for example,
+-- | to attach IDs: `mapWithAccum (\x i -> Tuple (i + 1) (Tuple x i)) 0`.
+mapWithAccum :: forall a b c. (a -> b -> Tuple b c) -> Event a -> b -> Event c
+mapWithAccum f xs acc = filterMap snd
+  $ fold (\f' -> map pure <<< f' <<< fst) (f <$> xs)
+  $ Tuple acc Nothing
+
+-- | When the first event is false, mute the second event.
+when :: Event Boolean -> Event ~> Event
```

Comment:
That's what it does do; sorry, I just worded the comment bizarrely! NT removed, too :)